src/processor/operator/recursive_extend/path_scanner.cpp

```diff
@@ -86,14 +87,14 @@ void PathScanner::initDfs(common::offset_t offset, size_t currentDepth) {
 }
 
 void PathScanner::writePathToVector(common::table_id_t tableID, common::ValueVector* pathVector,
-    common::ValueVector* dstNodeIDVector, common::sel_t& offsetVectorPos,
-    common::sel_t& dataVectorPos) {
+    common::ValueVector* dstNodeIDVector, common::ValueVector* pathLengthVector,
+    common::sel_t& offsetVectorPos, common::sel_t& dataVectorPos) {
+    assert(offsetVectorPos < common::DEFAULT_VECTOR_CAPACITY);
     auto pathLength = path.size();
     auto listEntry = common::ListVector::addList(pathVector, pathLength);
     pathVector->setValue(offsetVectorPos, listEntry);
-    dstNodeIDVector->setValue<common::nodeID_t>(
-        offsetVectorPos, common::nodeID_t{path[pathLength - 1], tableID});
-    assert(offsetVectorPos < common::DEFAULT_VECTOR_CAPACITY);
+    assert(path[pathLength - 1] == currentDstOffset);
+    writeDstNodeOffsetAndLength(dstNodeIDVector, pathLengthVector, offsetVectorPos, tableID);
```

Comment:
I'm a bit confused about the PR. so it looks like regardless of whether there is or there is not len(p) type expression, you always write path lengths to a value vector. Is that correct?

I was expecting to see a flag that you insert into these operators that will decide whether to write the lengths or not.

Reply:
Yes I always write path length to a vector because it's cost is same as writing dst node offsets.

Trying to be smart about writing length is doable but codes becomes a lot complicated after I made the change because we need to switch for track `path & length, path & !length, !path & length, !path & !length` in multiple places. I decide to go for readability and always write length for now.

Using enum is fine. I've moved from bool to enum.